File: backend/Models/Session.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PeerGrid.Backend.Models
{
    public class Session
    {
        [Key]
        public int Id { get; set; }

        public int? TutorId { get; set; }
        [ForeignKey("TutorId")]
        public User Tutor { get; set; }

        public int LearnerId { get; set; }
        [ForeignKey("LearnerId")]
        public User Learner { get; set; }

        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Topic { get; set; } = string.Empty;

        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public string Status { get; set; } = "Pending"; // Open, Pending, Active, Completed, Cancelled
        public decimal Cost { get; set; }
    }
}

File: backend(Spring Boot)/src/main/java/com/peergrid/backend/entity/Session.java
package com.peergrid.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.math.BigDecimal;

@Data
@Entity
@Table(name = "sessions")
public class Session {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "tutor_id")
    private User tutor;

    @ManyToOne
    @JoinColumn(name = "learner_id")
    private User learner;

    private String title = "";
    private String description = "";
    private String topic = "";

    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private String status = "Pending"; // Open, Pending, Active, Completed, Cancelled
    private BigDecimal cost;
}
